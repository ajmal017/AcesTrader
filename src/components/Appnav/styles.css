/* Appnav/styles.css */

.navbar-fixed {
  width: 100%;
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  z-index: 1000;
  padding-left: 0px;
  margin-left: 0px;
}

/*Navbar.Header {
    margin-top: 10px;
}*/

/*nav*/
.container {
  margin-left: 0px;
  /*margin-left: auto;*/
}

/* override bootstrap's margin 0f -15px to keep spinner on screen */
.navbar-right {
  margin-right: 0;
}

.navbar {
  margin-bottom: 0; /*overrides bootstrap*/
}

/*overrides bootstrap*/
.navbar-brand {
  font-size: 14px;
  list-style-type: none;
}

.navbar-nav li.loader {
  /* margin: 12px 6px 0 6px; */
  margin: 12px 0px 0 0px;
  visibility: hidden;
}

.navbar-nav li.loader.active {
  visibility: visible;
}

.navbar-logo {
  /* font-family: Georgia, 'Helvetica Neue', Helvetica, Arial, sans-serif; */
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: bold;
  font-size: 20px;
  padding-left: 20px;
  padding-right: 20px;
  color: #f8f9fa;
}

/*overrides bootstrap*/
/* padding: 10px 15px; */
.nav > li > a {
  padding: 16px 10px 4px 10px;
  font-size: 16px;
}
