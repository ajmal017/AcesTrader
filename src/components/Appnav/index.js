import React from 'react'
import { LinkContainer } from 'react-router-bootstrap'
import { Nav, Navbar, NavItem, NavDropdown, MenuItem } from 'react-bootstrap'
import './styles.css'

// class Appnav extends Component {
const Appnav = function() {
  return (
    <div>
      <Navbar className="navbar-fixed" inverse collapseOnSelect>
        <Navbar.Header>
          <Navbar.Brand>
            <LinkContainer to="/home">
              <NavItem eventKey={0}>
                <span className="navbar-logo">AcesTrader</span>
              </NavItem>
            </LinkContainer>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav>
            <NavDropdown eventKey={5} title="Prospects" id="5-basic-nav-dropdown">
              <LinkContainer to="/prospectbuys">
                <MenuItem eventKey={5.1}>Trading Buys</MenuItem>
              </LinkContainer>
              <LinkContainer to="/prospectsells">
                <MenuItem eventKey={5.2}>Trading Sells</MenuItem>
              </LinkContainer>
              <LinkContainer to="/prospecttrendfollowers">
                <MenuItem eventKey={5.2}>Trend Followers</MenuItem>
              </LinkContainer>
            </NavDropdown>

            <NavDropdown eventKey={4} title="Positions" id="5-basic-nav-dropdown">
              <LinkContainer to="/positionlongs">
                <MenuItem eventKey={4.1}>Trading Longs</MenuItem>
              </LinkContainer>
              <LinkContainer to="/positionshorts">
                <MenuItem eventKey={4.2}>Trading Shorts</MenuItem>
              </LinkContainer>
              <LinkContainer to="/positiontrendfollowers">
                <MenuItem eventKey={4.6}>Trend Followers</MenuItem>
              </LinkContainer>
            </NavDropdown>

            <NavDropdown eventKey={6} title="Symbols" id="5-basic-nav-dropdown">
              <LinkContainer to="/managebuyprospects">
                <MenuItem eventKey={6.2}>Trading Buy Prospects</MenuItem>
              </LinkContainer>
              <LinkContainer to="/managesellprospects">
                <MenuItem eventKey={6.6}>Trading Sell Prospects</MenuItem>
              </LinkContainer>
              <LinkContainer to="/managetrendfollowerprospects">
                <MenuItem eventKey={7.3}>Trend Follower Prospects</MenuItem>
              </LinkContainer>
            </NavDropdown>

            <NavDropdown eventKey={7} title="More" id="5-basic-nav-dropdown">
              <LinkContainer to="/scoreboard">
                <NavItem eventKey={7.1}>ScoreBoard</NavItem>
              </LinkContainer>

              <LinkContainer to="/peek">
                <NavItem eventKey={7.2}>Peek</NavItem>
              </LinkContainer>
              <MenuItem divider />

              <LinkContainer to="/home">
                <MenuItem eventKey={7.3}>About AcesTrader</MenuItem>
              </LinkContainer>
            </NavDropdown>
            <LinkContainer to="/alerts">
              <NavItem eventKey={9}>Alerts</NavItem>
            </LinkContainer>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </div>
  )
}
export default Appnav
