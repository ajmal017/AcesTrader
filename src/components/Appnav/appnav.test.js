import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Appnav from "../Appnav";

describe ('Appnav', () => {
  
  it('renders a Snapshot without error', () => {
    const tree = renderer.create(
      <Appnav />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });  

});

