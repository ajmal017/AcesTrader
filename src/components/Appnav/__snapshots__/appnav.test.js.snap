// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appnav renders a Snapshot without error 1`] = `
<div>
  <nav
    className="navbar-fixed navbar-inverse navbar navbar-default">
    <div
      className="container">
      <ul
        className="nav navbar-nav"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            action="push"
            href="#"
            onClick={[Function]}
            role="button">
            <span
              className="navbar-logo">
              Money Plan
            </span>
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href="#"
            id="2-basic-nav-dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            File
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="2-basic-nav-dropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                New
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Open...
              </a>
            </li>
            <li
              className="divider"
              onKeyDown={[Function]}
              role="separator"
              style={undefined} />
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Save
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Save As...
              </a>
            </li>
            <li
              className="divider"
              onKeyDown={[Function]}
              role="separator"
              style={undefined} />
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Close App
              </a>
            </li>
          </ul>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href="#"
            id="5-basic-nav-dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            Tools
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="5-basic-nav-dropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Request Trace...
              </a>
            </li>
            <li
              className="disabled"
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Read Trace Log
              </a>
            </li>
            <li
              className="divider"
              onKeyDown={[Function]}
              role="separator"
              style={undefined} />
            <li
              className="disabled"
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Read Error Log
              </a>
            </li>
          </ul>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            action="push"
            href="#"
            onClick={[Function]}
            role="button">
            Plan View
          </a>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            action="push"
            href="#"
            onClick={[Function]}
            role="button">
            Charts
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href="#"
            id="6-basic-nav-dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            Help
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="6-basic-nav-dropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Examples...
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Documentation...
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                Read Me
              </a>
            </li>
            <li
              className="divider"
              onKeyDown={[Function]}
              role="separator"
              style={undefined} />
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                action="push"
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                About Money Plan
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
